catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -66,147 +131,867 @@ public FederationAdminServiceImpl() {
+            throw new FederationAdminException(
+                    "Error updating registry entry. Metacard registry-id is not set.");
+        }
+        String registryId = updateMetacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+        List<Filter> filters = new ArrayList<>();
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(RegistryObjectMetacardType.REGISTRY_ID),
+                registryId));
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                RegistryConstants.REGISTRY_TAG));
+
+        List<Metacard> existingMetacards =
+                getRegistryMetacardsByFilter(FILTER_FACTORY.and(filters));
+
+        if (CollectionUtils.isEmpty(existingMetacards)) {
+            String message = "Error updating registry entry. Registry metacard not found.";
+            LOGGER.error("{} Registry metacard ID: {}", message, updateMetacard.getId());
+            throw new FederationAdminException(message);
+        }
+
+        if (existingMetacards.size() > 1) {
+            String message = "Error updating registry entry. Multiple registry metacards found.";
+
+            List<String> metacardIds = new ArrayList<>();
+            metacardIds.addAll(existingMetacards.stream()
+                    .map(Metacard::getId)
+                    .collect(Collectors.toList()));
+            LOGGER.error("{} Matching registry metacard ids: {}", message, metacardIds);
+
+            throw new FederationAdminException(message);
+        }
+
+        Metacard existingMetacard = existingMetacards.get(0);
+        boolean transientUpdate = false;
+        for (String transientAttributeKey : RegistryObjectMetacardType.TRANSIENT_ATTRIBUTES) {
+            Attribute transientAttribute = updateMetacard.getAttribute(transientAttributeKey);
+            if (transientAttribute == null) {
+                transientAttribute = existingMetacard.getAttribute(transientAttributeKey);
+                if (transientAttribute != null) {
+                    updateMetacard.setAttribute(transientAttribute);
+                }
+            } else {
+                Attribute existingAttribute = existingMetacard.getAttribute(transientAttributeKey);
+                if (existingAttribute == null) {
+                    transientUpdate = true;
+                    continue;
+                }
+                if (transientAttribute.getValues()
+                        .size() != existingAttribute.getValues()
+                        .size()) {
+                    transientUpdate = true;
+                    continue;
+                }
+                for (int i = 0; i < transientAttribute.getValues()
+                        .size(); i++) {
+                    if (!transientAttribute.getValues()
+                            .get(i)
+                            .equals(existingAttribute.getValues()
```

Comment:
There is a check above this loop that makes sure that existingAttribute.getValues() and transientAttribute.getValues() are the same size.